File: tourism/src/main/java/com/tourism/BookingService.java
package com.tourism;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;

import com.google.inject.Inject;

/**
 * Сервіс для роботи з даними про позиції бронювання в базу даних
 */
public class BookingService {
    private Connection connection;

    /**
     * Конструктор з впровадженням залежності від драйвера бази даних.
     *
     * @param connection з'єднання з базою даних
     */
    @Inject
    public BookingService(Connection connection) {
        this.connection = connection;
    }

    /**
     * Зберігає позицію бронювання в таблицю booking_items
     *
     * @param bookingItem позиція бронювання для збереження
     */
    public void saveBookingItem(BookingItem bookingItem) {
        String sql = "INSERT INTO booking_items (room_type, price) VALUES (?, ?)";

        try (PreparedStatement statement = connection.prepareStatement(sql)) {
            statement.setString(1, bookingItem.getRoomType());
            statement.setDouble(2, bookingItem.getPrice());
            statement.executeUpdate();
            System.out.println("✓ Позиція бронювання збережена в базу даних: " + 
                             bookingItem.getRoomType() + " | " + bookingItem.getPrice() + "$");
        } catch (SQLException e) {
            throw new RuntimeException("Failed to save booking item", e);
        }
    }

    /**
     * Повертає список всіх позицій бронювання з бази даних
     *
     * @return список позицій бронювання
     */
    public List<BookingItem> getAllBookingItems() {
        List<BookingItem> bookingItems = new ArrayList<>();
        String sql = "SELECT room_type, price FROM booking_items";

        try (PreparedStatement statement = connection.prepareStatement(sql);
             ResultSet resultSet = statement.executeQuery()) {
            
            while (resultSet.next()) {
                String roomType = resultSet.getString("room_type");
                double price = resultSet.getDouble("price");
                bookingItems.add(new BookingItem(roomType, price));
            }
        } catch (SQLException e) {
            throw new RuntimeException("Failed to retrieve booking items", e);
        }

        return bookingItems;
    }
}

File: tourism-python/src/tourism/booking_service.py
from tourism.booking_item import BookingItem


class BookingService:
    """
    Сервіс для збереження даних про позиції бронювання в БД
    """

    def __init__(self, connection):
        self.connection = connection

    def save_booking_item(self, booking_item: BookingItem):
        sql = """
        INSERT INTO booking_items (room_type, price)
        VALUES (?, ?)
        """
        cursor = self.connection.cursor()
        cursor.execute(sql, (
            booking_item.get_room_type(),
            booking_item.get_price()
        ))
        self.connection.commit()

        print(
            f"✓ Позиція бронювання збережена в БД: "
            f"{booking_item.get_room_type()} | {booking_item.get_price()}$"
        )
